src/main/scala-2.11/markets/participants/LiquidityDemander.scala

```diff
@@ -15,53 +15,40 @@ limitations under the License.
 */
 package markets.participants
 
-import akka.actor.Scheduler
 
-import markets.orders.Order
+import markets.orders.market.{MarketAskOrder, MarketBidOrder}
+import markets.tradables.Tradable
 
-import scala.concurrent.ExecutionContext
-import scala.concurrent.duration.FiniteDuration
 
-
-/** A Trait providing behavior necessary to submit `MarketOrderLike` orders. */
 trait LiquidityDemander extends MarketParticipant {
 
-  def generateMarketOrder(): Order
+  def marketOrderTradingStrategy: MarketOrderTradingStrategy
 
-  /** Schedule a market order.
-    *
-    * @param scheduler
-    * @param initialDelay
-    * @param executionContext
-    */
-  def scheduleMarketOrder(scheduler: Scheduler,
-                          initialDelay: FiniteDuration)
-                         (implicit executionContext: ExecutionContext): Unit = {
-    scheduler.scheduleOnce(initialDelay, self, SubmitMarketOrder)(executionContext)
+  override def receive: Receive = {
+    case SubmitMarketAskOrder =>
+      marketOrderTradingStrategy.askOrderStrategy(tickers) match {
+        case Some((quantity, tradable)) =>
+          val marketAskOrder = generateMarketAskOrder(quantity, tradable)
+          submit(marketAskOrder)
+        case None =>  // No feasible marketAskOrderStrategy!
+      }
+    case SubmitMarketBidOrder =>
+      marketOrderTradingStrategy.bidOrderStrategy(tickers) match {
+        case Some((quantity, tradable)) =>
+          val marketBidOrder = generateMarketBidOrder(quantity, tradable)
+          submit(marketBidOrder)
+        case None =>  // No feasible marketBidOrderStrategy!
+      }
+    case message => super.receive(message)
   }
 
-  /** Schedule a market order.
-    *
-    * @param scheduler
-    * @param initialDelay
-    * @param interval
-    * @param executionContext
-    */
-  def scheduleMarketOrder(scheduler: Scheduler,
-                          initialDelay: FiniteDuration,
-                          interval: FiniteDuration)
-                         (implicit executionContext: ExecutionContext): Unit = {
-    scheduler.schedule(initialDelay, interval, self, SubmitMarketOrder)(executionContext)
+  private final def generateMarketAskOrder(quantity: Long, tradable: Tradable) = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [generateMarketAskOrder should have a smaller scope by using private[this]](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1704906043/issues/source?bid=2921361&fileBranchId=3046267#l45)